packages/core/http/core-http-router-server-internal/src/request.ts

```diff
@@ -73,13 +81,36 @@ export class CoreKibanaRequest<
     withoutSecretHeaders: boolean = true
   ) {
     let requestParts: { params: P; query: Q; body: B };
-    if (routeSchemas === undefined || isFakeRawRequest(req)) {
-      requestParts = { query: {} as Q, params: {} as P, body: {} as B };
+
+    if (isFakeRawRequest(req)) {
+      requestParts = {
+        query: {} as Q,
+        params: {} as P,
+        body: {} as B,
+      };
+
+      return new CoreKibanaRequest(
+        req,
+        requestParts.params,
+        requestParts.query,
+        requestParts.body,
+        withoutSecretHeaders
+      );
+    }
+
+    const rawParts = sanitizeRequest(req);
```

Comment:
I think this belongs in the `else` block below.

Reply:
Reverted it to the exact code we have in `main` in 8603a52f864286a327d6aea5bd392cbf217a3a38